providers/gcs/gcs.go

```diff
@@ -135,8 +136,15 @@ func appendHttpOptions(gc Config, opts []option.ClientOption) ([]option.ClientOp
 		return nil, err
 	}
 
-	httpCli := &http.Client{
+	hedgedClient, err := hedgedhttp.New(hedgedhttp.Config{
```

Comment:
All of this `hedgedhttp.New()` must be inside Thanos code, not here, because otherwise you will be forcing hedged HTTP requests on all clients of this library.

Reply:
Thank you for your assistance ! i want to confirm my query that you want me to create hedgedhttp.Client into the Thanos codebase itself, and expose a configuration option only to enable or disable hedged requests? something like this ?